qiskit/providers/backend.py

```diff
@@ -229,3 +233,239 @@ class can handle either situation.
+        pass
+
+    @property
+    @abstractmethod
+    def num_qubits(self) -> int:
+        """Return the number of qubits the backend has."""
+        pass
+
+    @property
+    def coupling_map(self):
+        """Return the :class:`~qiskit.transpiler.CouplingMap` object"""
+        return self.target.coupling_map()
+
+    @property
+    def instruction_durations(self):
+        """Return the :class:`~qiskit.transpiler.InstructionDurations` object."""
+        return self.target.durations()
+
+    @property
+    @abstractmethod
+    def conditional(self) -> bool:
+        """Return bool whether the target can execute gates with classical
+        conditions."""
+        pass
+
+    @property
+    @abstractmethod
+    def max_shots(self) -> int:
+        """Return the maximum number of shots supported by the backend.
+
+        If there is no limit this will return None
+        """
+        pass
+
+    @property
+    @abstractmethod
+    def max_circuits(self):
+        """The maximum number of circuits (or Pulse schedules) that can be
+        run in a single job.
+
+        If there is no limit this will return None
+        """
+        pass
+
+    @classmethod
+    @abstractmethod
+    def _default_options(cls):
+        """Return the default options
+
+        This method will return a :class:`qiskit.providers.Options`
+        subclass object that will be used for the default options. These
+        should be the default parameters to use for the options of the
+        backend.
+
+        Returns:
+            qiskit.providers.Options: A options object with
+                default values set
+        """
+        pass
+
+    def t1(self, qubit: Union[int, List[int]]) -> Union[float, List[float]]:
+        """Return the T1 time of a given qubit
+
+        Args:
+            qubit: The qubit index to get the T1 time for. If
+                a list is specified the output will be a list with the
+                T1 time for the specified qubits in the same order.
+
+        Returns:
+            t1: the T1 time for the specified qubit(s) in seconds.
+
+        Raises:
+            NotImplementedError: if the backend doesn't support querying the
+                t1 time for a qubit
+        """
+        raise NotImplementedError
+
+    def t2(self, qubit: Union[int, List[int]]) -> Union[float, List[float]]:
+        """Return the T2 time of a given qubit
+
+        Args:
+            qubit: The qubit index or indices to get the T2 time for. If
+                a list is specified the output will be a list with the
+                T2 time for the specified qubits in the same order.
+
+        Returns:
+            t2: the T2 time for the specified qubit(s) in seconds
+
+        Raises:
+            NotImplementedError: if the backend doesn't support querying the
+                t2 time for a qubit
+        """
+        raise NotImplementedError
+
+    def qubit_frequency(self, qubit: Union[int, List[int]]) -> Union[float, List[float]]:
```

Comment:
I think t1 and t2 are useful to the noise-sensitive transpilation or at least selection of qubits for the initial layout, however frequency is not useful information to run programs (unless you do collision analysis). If we provide qubit frequency, this should come with anharmonicity and resonator frequency. Perhaps we can provide `Hamiltonian` object as well as `Target`.

Reply:
That's a good point, I'll remove frequency from here and we can add this information to a Hamiltonian object

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/5885/commits/154eee701091b92886dfff01cba3408a0c678c55